docs/DeveloperGuide.md

```diff
@@ -4,10 +4,137 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```

Comment:
Good that the acknowledgment section is here. But will it be better to update the section?